pkg/controller/atlasdeployment/searchindex_test.go

```diff
@@ -697,8 +697,50 @@ func Test_searchIndexReconciler(t *testing.T) {
 			ID:                         pointer.MakePtr("testID"),
 			Status:                     nil,
 		}
-		result := reconciler.reconcileInternal(idxInAKO, idxInAtlas)
+		result := reconciler.reconcileInternal("", idxInAKO, idxInAtlas)
 		assert.False(t, result.IsOk())
 		assert.True(t, reconciler.ctx.HasReason(status.SearchIndexStatusError))
 	})
+
+	t.Run("DROP: Must clear if the index disappeared from Atlas", func(t *testing.T) {
+		mockSearchAPI := mockadmin.NewAtlasSearchApi(t)
+		atlasSearch := searchindex.NewProductionAtlasSearch(mockSearchAPI)
+
+		testCluster := &akov2.AtlasDeployment{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testDeployment",
+				Namespace: "testNamespace",
+			},
+			Spec: akov2.AtlasDeploymentSpec{
+				DeploymentSpec: &akov2.AdvancedDeploymentSpec{
+					Name: "testDeploymentName",
+				},
+			},
+			Status: status.AtlasDeploymentStatus{},
+		}
+
+		reconciler := &searchIndexReconciler{
+			ctx: &workflow.Context{
+				Log:     zap.S(),
+				OrgID:   "testOrgID",
+				Context: context.Background(),
+			},
+			deployment:    testCluster,
+			k8sClient:     nil,
+			projectID:     "",
+			indexName:     "testIndexName",
+			searchService: atlasSearch,
+		}
+		// In previous status, but not found in Atlas now
+		idxInAtlas := &searchindex.SearchIndex{
```

Comment:
this test is missing the case where `idxInAtlas` is empty which can happen.

Reply:
tested